### PR TITLE
Ensure latest version is selected on docs page load

### PR DESCRIPTION
**Pull request by nayres**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-na.add-release-stage-to-version-hashicorp.vercel.app) 🔎
- [Asana task](https://app.asana.com/0/1203347061500809/1203374761617827) 🎟️

## 🗒️ What

This PR ensures the default selected version on a docs page is the latest version based on the `isLatest` value from the Content API. This allows pre-released docs versions to be present in the version selector, but not the default version displayed on docs page load. 

## 🤷 Why

This is a UI task that is part of a larger effort to support docs pre-releases. See [RFC](https://docs.google.com/document/d/1yPrfcQCaEZyyLii8hv-OmoYScqpg94_OAfaFXymo7Y4) for more information.

## 📸 Design Screenshots

[UI design](https://docs.google.com/document/d/1yPrfcQCaEZyyLii8hv-OmoYScqpg94_OAfaFXymo7Y4/edit#heading=h.x7zzzrdfycxx)

## 🧪 Testing

- [ ] Checkout this branch
- [ ] Run `npm install`
- [ ] Run `npm run start`
- [ ] Navigate to any product doc page. For example: `localhost:3000/boundary/docs`
- [ ] Interact with the version selector in the top right of the docs content. You will notice a version tagged `beta | alpha | rc`

### CODE REVIEW

File: src/components/docs-version-switcher/docs-version-switcher.test.tsx [hunk 1]
```diff
@@ -67,7 +87,7 @@ describe('DocsVersionSwitcher', () => {
 
 		// assert that only 2 of the 3 versions are shown in the dropdown
```
Comment (thiskevinwang) [inline src/components/docs-version-switcher/docs-version-switcher.test.tsx:88]:
```suggestion
		// assert that only n-1 versions are shown in the dropdown
```

File: .env [hunk 2]
```diff
@@ -9,7 +9,8 @@ NEXT_PUBLIC_ALGOLIA_SEARCH_ONLY_API_KEY=2cd9898a6c253bfa3965d2b62a4f7f3d
 # which takes precedence over the NEXT_PUBLIC_ALGOLIA_INDEX env var.
 # NEXT_PUBLIC_ALGOLIA_INDEX=product_WAYPOINT
 
-MKTG_CONTENT_API="https://content.hashicorp.com"
+# MKTG_CONTENT_API="https://content.hashicorp.com"
+MKTG_CONTENT_API="https://mktg-content-api-git-pre-release-api-stub-hashicorp.vercel.app"
```
Comment (thiskevinwang) [inline .env:13]: Be sure to undo this change

**Review by thiskevinwang — DISMISSED** [review]
Looks good to me! Let a few tiny comments.

Let me know when you're ready to move this out of **Draft** status, and I can give this an official ✅ !

File: .env (hunk 2 above)
Comment (nayres) [reply .env:13]: Yeah this is just temporary to allow reviewers to test locally. I'll undo this once it looks good to go.

File: package.json [hunk 3]
```diff
@@ -114,7 +114,7 @@
 		"@hashicorp/react-code-block": "^6.2.2",
 		"@hashicorp/react-command-line-terminal": "^3.0.0",
 		"@hashicorp/react-consent-manager": "^9.0.1",
-		"@hashicorp/react-docs-page": "^17.7.2",
+		"@hashicorp/react-docs-page": "^17.8.0-canary-20230411190718",
```
Comment (thiskevinwang) [inline package.json:117]: @nayres Let's get publish `17.8.0` (stable) published, update this, and then this PR should be good to go!

**Review by thiskevinwang — APPROVED** [review]
LGTM! - Don't forget to update the `@hashicorp/react-docs-page` canary version.

Feel free to merge after!
